### DISCUSSION TITLE
AmirKabir University of Technology AP1400-2: Advanced Programming - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#311] [Announcements]
# AmirKabir University of Technology AP1400-2: Advanced Programming - CS自学指南

CS自学指南

https://csdiy.wiki/%E7%BC%96%E7%A8%8B%E5%85%A5%E9%97%A8/AUT1400/

**Comment by hsj3860** [comment]
有人做这个课程的作业嘛？CS106L今年的assignment真的挺垃圾的，准备用这个代替一下。
如果有人也有做这个的话，可以加wx或者qq交流一下

**Reply by GongNanyue** [reply]
这个不同时间的作业还是不一样的吗

**Reply by AlwaysBluer** [reply]
请问这个是要docker build以后测评吗，还是在docker 里面完成代码

**Reply by AlwaysBluer** [reply]
大佬能交流一下吗

**Reply by tang-jiapeng** [reply]
做了第一个

**Reply by zippermonkey** [reply]
刚在做第一个，我是自己装了下GTEST就没用docker了 测试正常

**Reply by Fontzs** [reply]
106L的作业确实垃圾

**Reply by calyle** [reply]
直接在docker里面完成代码吧，所有的proj都是相同的docker环境

**Comment by zippermonkey** [comment]
建了个QQ群，723678120

**Comment by Fontzs** [comment]
https://github.com/Fontzs/AUT1400 
第7个不想写，第2个还有最后一个测试过不了，其他的全通过测试了

**Reply by Daniel741312** [reply]
size_t Client::generate_nonce() {
    return e() % 10;
}

**Reply by Daniel741312** [reply]
这里不应该返回一个范围很小的随机数，因为只有哈希碰撞次数足够多，才有可能在前10个字符中出现“000”

**Comment by Daniel741312** [comment]
贴一个自己的实现：https://github.com/Daniel741312/AUT-1400

**Reply by DoubleQiany** [reply]
very nice job!

**Comment by thinklive1** [comment]
如果想继续用cs106b的qt做，只要用qt编译一下最新版本的gtest里面的gtest和gmock，并在作业的cmakelist里把这两个库的路径包含进去就行了

**Comment by 0130w** [comment]
https://github.com/0130w/AP1400-2
贴一个自己的前6个的实现(

**Comment by wobushicsy** [comment]
这个难度五星有点高了，个人认为只到三星的难度（大约和61B相同）

**Comment by pnyyb** [comment]
请问这个作业是怎么做啊，第一次用github

**Reply by shabi123231** [reply]
我也不会用这个，你现在知道怎么用吗

**Reply by sbwcwso** [reply]
> 我也不会用这个，你现在知道怎么用吗

可以参考我的笔记  [C++/Advanced Programming Course](https://note.sbwcwso.com/CSStudy/#/page/c%2B%2B%2Fadvanced%20programming%20course) ，主要是 docker 的配置。配好之后可以在 vscode 中直接连到 container 中进行使用。

**Reply by QWWQaw** [reply]
看了但是不明白，docker配置该怎么搞

**Reply by QWWQaw** [reply]
搞明白了

**Comment by sbwcwso** [comment]
个人实现: [C++/Advanced Programming Course
](https://note.sbwcwso.com/CSStudy/#/page/c%2B%2B%2Fadvanced%20programming%20course)
欢迎讨论交流😁

**Reply by cangahai** [reply]
请问从哪里可以看这个课程呀

**Reply by sbwcwso** [reply]
> 请问从哪里可以看这个课程呀

没有课程，只有试验。课程如果有其它语言基础的话，可以看 CS106L

**Reply by QWWQaw** [reply]
大佬，那个docker我启动镜像失败

**Reply by QWWQaw** [reply]
docker: Error response from daemon: create .: volume name is too short, names should be at least two alphanumeric characters.它报了这个错

**Reply by cse24lnab** [reply]
安装这个docker是可以类似oj那样测试吗

**Reply by q3487633796** [reply]
佬，笔记失效了；请问这个作业要怎么做啊？github用得不多

**Comment by Andy-xiaokang** [comment]
我也来贴一个我的 [C++ Route Map](https://github.com/Andy-xiaokang/AP1400-2/tree/master), 是 CS106B, CS106L, AP1400-2 的综合

**Reply by cangahai** [reply]
请问从哪里可以看这个课程呀

**Comment by sbwcwso** [comment]


**Reply by cangahai** [reply]
请问从哪里可以看这个课程呀

**Reply by dawushi97** [reply]
这个是作业没有课程，github上和评论区中有很多实现，你可以参考一下

**Comment by Chena2003** [comment]
贴一个我的实现：<https://github.com/Chena2003/AUT-AP1400-2>
欢迎同学们参考😀😀😀

**Comment by lipengpai** [comment]
一直显示找不到gtest.h源文件,有佬佬知道怎么解决吗

**Comment by lipengpai** [comment]
你好,虽然不认识你,我也不是佬(悲),但还是问一句问题解决了吗?
直接在github上搜索对应的仓库,然后git clone一下就可以了(具体细节问问ai,不难的)
然后如果你要做的话(因为我没做),建议使用clion(vs上的环境我一直没配好,这才放弃了的)
总之,祝你顺利🥰

**Reply by YLYCTS** [reply]
虽然已经过去四个月了，但是我还是想告诉你，每一个作业都配备了Dockerfile，所以不需要自己亲自配置环境，直接使用docker就行，如果有报错信息，直接粘贴给ai就可以解决
